pkg/tsdb/influxdb/query.go

```diff
@@ -78,7 +78,29 @@ func (query *Query) renderTags() []string {
 	return res
 }
 
+func isTimeRangeNumeric(tr *plugins.DataTimeRange) bool {
+	if _, err := strconv.ParseInt(tr.From, 10, 64); err != nil {
+		return false
+	}
+	if _, err := strconv.ParseInt(tr.To, 10, 64); err != nil {
+		return false
+	}
+	return true
+}
+
 func (query *Query) renderTimeFilter(queryContext plugins.DataQuery) string {
+	// If from expressions
+	if isTimeRangeNumeric(queryContext.TimeRange) {
+		from, to, err := epochMStoInfluxTime(queryContext.TimeRange)
```

Comment:
a minor remark: `isTimeRangeNumeric` and `epochMStoInfluxTime` are pretty similar so maybe the condition could change and check if the `epochMStoInfluxTime` returns an error to fall back to non-expression time range processing.